src/app.py

```diff
@@ -3,30 +3,67 @@
-    assigning them to execute on a second pool
+        * Front Office Position
+        * Back Office Position
+        * Depot Position
+        * Cash Balance
+        * Cash Flow
+        * Settlement Instruction
+
+    Domain objects are created sequentially, with one domain object having all
+    output files written before starting on the next domain object. This is
+    sequenced to ensure that inter-object dependencies are created correctly.
+
+    Dependencies are established using a local SQLite database file. When
+    creating a domain object, any fields that may be referenced by dependant
+    objects are persisted to the database. Upon creating a dependant object,
+    the database is queried to retrieve an appropriate value.
+
+    When creating domain object records and writing them to files,
+    multiprocessing is used to increase time efficiency. A Coordinator object
+    is responsible for creating and managing these processes, and it will
+    spawn two parent processes: 'create_parent_process' and
+    'write_parent_process'. These each spawn a number of child processes as
+    specified in the 'shared_args' section of the user config.
+
+    The Coordinator will populate a FIFO 'create_job_queue' with create jobs
+    (dictionaries specifying a quantity of domain object records to be created
+    and later written to output file). The maximum quantity of records in a
+    create job is specified in the 'shared_args' section of the user config.
+
+    The create parent process will wait until the create job queue is not
+    empty, at which point it will dequeue create jobs and add them to a list.
+    It will not dequeue more than double the number of child create processes
+    worth of create jobs at a time to ensure that the child processes work on
+    reasonable sized batches of create jobs.
+
+    The dequeued create jobs in the list are executed over a pool of child
+    create processes, which, upon termination of all child processes, returns
+    a list created records. This list contains the collated output from that
+    batch of create jobs.
+
+    As batches of create jobs are run, the returned lists of records are added
+    to a FIFO 'generated_record_queue'. This queue is shared between the the
+    create and write parent processes.
+
+    The write parent process waits until the 'generated_record_queue' is not
+    empty, at which point it retrieves lists of records from the queue and
+    collates all records from those lists into a
+    'dequeued_created_records_not_yet_written_to_file' list.
+
+    It then creates a list of write jobs, each specifying a quantity of records
+    from the 'dequeued_created_records_not_yet_written_to_file' list to write
+    to file, and the name of the output file. A single output file can have
+    multiple write jobs, but a write job can only refer to one output file.
+
+    The write jobs in the list are run over a pool of child write processes,
```

Comment:
I think this paragraph is useful, but because it's quite a high level view on things I think it would be better included in the readme rather than in this class

Reply:
I have moved all multiprocessing implementation details into the Creator and Writer class docstrings since this seems to be the best fit for a high level implementation summary